src/core/server/integration_tests/saved_objects/migrations/kibana_migrator_test_kit.expect.ts

```diff
@@ -15,10 +15,11 @@ export async function expectDocumentsMigratedToHighestVersion(
   index: string | string[]
 ) {
   const typeMigrationVersions: Record<string, string> = {
-    basic: '10.1.0', // did not define any model versions
-    server: '10.1.0', // did not define any model versions
-    deprecated: '10.1.0', // did not define any model versions
+    basic: '10.1.0', // did not define any new model versions
+    server: '10.1.0', // did not define any new model versions
+    deprecated: '10.1.0', // did not define any new model versions
     complex: '10.2.0',
+    old: '8.9.0',
```

Comment:
Here we are ensuring that documents that only defined `migrations:` continue to work as they did before this PR, i.e. the `typeMigrationVersion` property stays at the latest migration defined.

That's why I introduced the `'old'` type in the testing data.